File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int n=nums.size();
        int major=nums[0];
        int count=1;
        for(int i=1;i<n;i++){
            if(count==0){
                major=nums[i];
                count++;
            }
            else if(nums[i]==major){
                count++;
            }
            else{
                count--;
            }
        }
        return major;
    }
};

File: 0169-majority-element/0169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        d={}
        ans=nums[0]
        n=len(nums)
        for i in nums:
            if i in d:
                d[i]+=1
                if d[i]>n/2:
                    ans=i
                    break
            else:
                d[i]=1
        return ans